### ISSUE TITLE
Error while deploy - InvalidParameterValue: Environment

### ISSUE THREAD

**Issue by AlexeyGavrilovST** [zodern/mup-aws-beanstalk#112]
Hello

I got error during deploy 
```
λ sh deploy.sh
okokloaded config from C:\work\parts\.deploy-staging\mup.js

1 Validation Errors
  - "proxy" app.env.PORT is ignored when using the reverse proxy

Read the docs and view example configs at
    http://meteor-up.com/docs

=> Setting up
=> Ensuring IAM Roles and Instance Profiles are setup
Building App Bundle Locally
Node#moveTo was deprecated. Use Container#append.
=> Archiving Bundle
  10% Archived
  20% Archived
  30% Archived
  40% Archived
  50% Archived
  60% Archived
  70% Archived
  80% Archived
  90% Archived
  100% Archived
=> Uploading bundle
  Uploaded 7%
  Uploaded 14%
  Uploaded 21%
  Uploaded 28%
  Uploaded 35%
  Uploaded 43%
  Uploaded 50%
  Uploaded 57%
  Uploaded 64%
  Uploaded 71%
  Uploaded 79%
  Uploaded 86%
  Uploaded 93%
  Uploaded 100%
  Finishing upload. This could take a couple minutes
=> Creating Version
=> Configuring Beanstalk Environment
InvalidParameterValue: Environment named mup-env-partsledger is in an invalid state for this operation. Must be Ready.
    at Request.extractError (C:\work\parts\node_modules\aws-sdk\lib\protocol\query.js:50:29)
    at Request.callListeners (C:\work\parts\node_modules\aws-sdk\lib\sequential_executor.js:106:20)
    at Request.emit (C:\work\parts\node_modules\aws-sdk\lib\sequential_executor.js:78:10)
    at Request.emit (C:\work\parts\node_modules\aws-sdk\lib\request.js:688:14)
    at Request.transition (C:\work\parts\node_modules\aws-sdk\lib\request.js:22:10)
    at AcceptorStateMachine.runTo (C:\work\parts\node_modules\aws-sdk\lib\state_machine.js:14:12)
    at C:\work\parts\node_modules\aws-sdk\lib\state_machine.js:26:10
    at Request.<anonymous> (C:\work\parts\node_modules\aws-sdk\lib\request.js:38:9)
    at Request.<anonymous> (C:\work\parts\node_modules\aws-sdk\lib\request.js:690:12)
    at Request.callListeners (C:\work\parts\node_modules\aws-sdk\lib\sequential_executor.js:116:18)
    at Request.emit (C:\work\parts\node_modules\aws-sdk\lib\sequential_executor.js:78:10)
    at Request.emit (C:\work\parts\node_modules\aws-sdk\lib\request.js:688:14)
    at Request.transition (C:\work\parts\node_modules\aws-sdk\lib\request.js:22:10)
    at AcceptorStateMachine.runTo (C:\work\parts\node_modules\aws-sdk\lib\state_machine.js:14:12)
    at C:\work\parts\node_modules\aws-sdk\lib\state_machine.js:26:10
    at Request.<anonymous> (C:\work\parts\node_modules\aws-sdk\lib\request.js:38:9)
```

What logs I see in aws console: Elastic Beanstalk ->Environments -> mup-env-partsledger -> Health

```
Application deployment failed at 2020-07-31T15:38:38Z with exit status 1 and error: Hook /opt/elasticbeanstalk/hooks/appdeploy/pre/50npm.sh failed.

+ /opt/elasticbeanstalk/containerfiles/ebnode.py --action npm-install

> mup-partsledger@1.0.0 preinstall /tmp/deployment/application
> cd programs/server && npm install

npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported

> fibers@2.0.0 install /tmp/deployment/application/programs/server/node_modules/fibers
> node build.js || nodejs build.js

make: Entering directory `/tmp/deployment/application/programs/server/node_modules/fibers/build'
CXX(target) Release/obj.target/fibers/src/fibers.o
../src/fibers.cc:68:2: error: ‘Handle’ does not name a type
Handle<String> NewLatin1String(Isolate* isolate, const char* string) {
^
../src/fibers.cc:72:2: error: ‘Handle’ does not name a type
Handle<String> NewLatin1Symbol(Isolate* isolate, const char* string) {
^
../src/fibers.cc:94:2: error: ‘Handle’ does not name a type
Handle<Object> NewInstance(Isolate* isolate, Local<Function> fn, int argc, Local<Value> argv[]) {
^
../src/fibers.cc:104:2: error: ‘Handle’ does not name a type
Handle<Number> ToNumber(Local<Value> value) {
^
../src/fibers.cc:126:58: error: ‘Handle’ has not been declared
void Reset(Isolate* isolate, Persistent<T>& persistent, Handle<T> handle) {
^
../src/fibers.cc:126:64: error: expected ‘,’ or ‘...’ before ‘<’ token
void Reset(Isolate* isolate, Persistent<T>& persistent, Handle<T> handle) {
^
../src/fibers.cc: In function ‘void uni::Reset(v8::Isolate*, v8::Persistent<T, v8::NonCopyablePersistentTraits<T> >&, int)’:
../src/fibers.cc:127:29: error: ‘handle’ was not declared in this scope
persistent.Reset(isolate, handle);
^
../src/fibers.cc: At global scope:
../src/fibers.cc:139:26: error: variable or field ‘SetInternalPointer’ declared void
void SetInternalPointer(Handle<T> handle, int index, void* val) {
^
../src/fibers.cc:139:26: error: ‘Handle’ was not declared in this scope
../src/fibers.cc:139:34: error: expected primary-expression before ‘>’ token
void SetInternalPointer(Handle<T> handle, int index, void* val) {
^
../src/fibers.cc:139:36: error: ‘handle’ was not declared in this scope
void SetInternalPointer(Handle<T> handle, int index, void* val) {
^
../src/fibers.cc:139:44: error: expected primary-expression before ‘int’
void SetInternalPointer(Handle<T> handle, int index, void* val) {
^
../src/fibers.cc:139:55: error: expected primary-expression before ‘void’
void SetInternalPointer(Handle<T> handle, int index, void* val) {
^
../src/fibers.cc:143:27: error: template declaration of ‘void* uni::GetInternalPointer’
void* GetInternalPointer(Handle<T> handle, int index) {
^
../src/fibers.cc:143:27: error: ‘Handle’ was not declared in this scope
../src/fibers.cc:143:35: error: expected primary-expression before ‘>’ token
void* GetInternalPointer(Handle<T> handle, int index) {
^
../src/fibers.cc:143:37: error: ‘handle’ was not declared in this scope
void* GetInternalPointer(Handle<T> handle, int index) {
^
../src/fibers.cc:143:45: error: expected primary-expression before ‘int’
void* GetInternalPointer(Handle<T> handle, int index) {
^
../src/fibers.cc:148:2: error: ‘Handle’ does not name a type
Handle<T> Deref(Isolate* isolate, Persistent<T>& handle) {
^
../src/fibers.cc:153:14: error: variable or field ‘Return’ declared void
void Return(Handle<T> handle, const Arguments& args) {
^
../src/fibers.cc:153:14: error: ‘Handle’ was not declared in this scope
../src/fibers.cc:153:22: error: expected primary-expression before ‘>’ token
void Return(Handle<T> handle, const Arguments& args) {
^
../src/fibers.cc:153:24: error: ‘handle’ was not declared in this scope
void Return(Handle<T> handle, const Arguments& args) {
^
../src/fibers.cc:153:32: error: expected primary-expression before ‘const’
void Return(Handle<T> handle, const Arguments& args) {
^
../src/fibers.cc:157:14: error: variable or field ‘Return’ declared void
void Return(Handle<T> handle, GetterCallbackInfo info) {
^
../src/fibers.cc:157:14: error: ‘Handle’ was not declared in this scope
../src/fibers.cc:157:22: error: expected primary-expression before ‘>’ token

```

This is my mup.js (I hide private info by *******):
```
module.exports = {
	servers: {
		one: {
			host: '*******',
			username: 'ubuntu',
			pem: '~/.aws/serviceskey.pem',
		},
		two: {
			host: '*******',
			username: 'ubuntu',
			pem: '~/.aws/serviceskey.pem'
		}
	},

	proxy: {
		domains: '*******',
	},

	app: {
		name: 'PartsLedger',
		path: '../',
		type: 'aws-beanstalk',

		auth: {
			id: '*******',
			secret: '*******'
		},

		minInstances: 2,
		maxInstances: 2,
		instanceType: 't2.medium',

		buildOptions: {
			serverOnly: true,
		},

		customBeanstalkConfig: [
			{
				namespace: 'aws:elasticbeanstalk:application',
				option: 'Application Healthcheck URL',
				value: '/healthcheck'
			},
			{
				namespace: 'aws:elasticbeanstalk:environment:process:default',
				option: 'HealthCheckPath',
				value: '/healthcheck'
			},
			{
				namespace: 'aws:elasticbeanstalk:environment:process:default',
				option: 'Protocol',
				value: 'HTTPS'
			},
			{
				namespace: 'aws:elasticbeanstalk:healthreporting:system',
				option: 'ConfigDocument',
				value: JSON.stringify({
					"Rules": {
						"Environment": {
							"Application": {
								"ApplicationRequests4xx": {
									"Enabled": true
								}
							},
							"ELB": {
								"ELBRequests4xx": {
									"Enabled": true
								}
							}
						}
					},
					"Version": 1
				})
			}
		],

		yumPackages: {
			'gcc-c++': ''
		},

		env: {
			ROOT_URL: '*******',
			MONGO_URL: '*******',
		},

		// docker: {
		// 	// change to 'kadirahq/meteord' if your app is using Meteor 1.3 or older
		// 	//image: 'abernix/meteord:base',
		// 	//image: 'abernix/meteord:node-8.4.0-base',
		// 	image: 'abernix/meteord:node-8-base',
		// }
	},
	plugins: ['mup-aws-beanstalk']
};

```

I tried with docker image, but it's not working:
`image: 'abernix/meteord:node-8-base'`

Also tried to remove gcc-c++ package.

Tried to add .ebextensions\swap.config
```
container_commands:
    01setup_swap:
        command: "bash .ebextensions/setup_swap.sh"
```
and .ebextensions\setup_swap.sh
```
#!/bin/bash

SWAPFILE=/var/swapfile
SWAP_MEGABYTES=2048

if [ -f $SWAPFILE ]; then
	echo "Swapfile $SWAPFILE found, assuming already setup"
	exit;
fi

/bin/dd if=/dev/zero of=$SWAPFILE bs=1M count=$SWAP_MEGABYTES
/bin/chmod 600 $SWAPFILE
/sbin/mkswap $SWAPFILE
/sbin/swapon $SWAPFILE
```

Still not working. 

Any ideas why it's not working?

Thanks for help!

**Comment by zodern** [comment]
In the next version it will wait until the environment has the correct state. If you go to https://console.aws.amazon.com/elasticbeanstalk/home?region=us-east-1#/environments and select the environment, it should say something like `Elastic Beanstalk is updating your environment.` Once that is done, deploys will work.

**Comment by zodern** [comment]
This will be fixed in 0.7 (#142 )
